src/vm/instruction_set.md

```diff
@@ -647,6 +679,363 @@ Panic if:
+
+Clears `$of` and `$err`.
+
+Panic if:
+
+- A reserved compare mode is given
+- `$rA` is a [reserved register](./index.md#semantics)
+- `$rB + 16` overflows or `> VM_MAX_RAM`
+- `indirect == 1` and `$rC + 16` overflows or `> VM_MAX_RAM`
+
+### WQCM: 256-bit integer comparison
+
+|             |                                                                                       |
+|-------------|---------------------------------------------------------------------------------------|
+| Description | Compare or examine two 256-bit integers using selected mode                           |
+| Operation   | `b = mem[$rB,32]`<br>;`c = indirect?mem[$rC,32]:$rC;`<br>`$rA = cmp_op(b,c);`         |
+| Syntax      | `wqcm $rA, $rB, $rC, imm`                                                             |
+| Encoding    | `0x00 rA rB rC i`                                                                     |
+| Notes       |                                                                                       |
+
+The immediate value is interpreted identically to `WDCM`.
+
+Clears `$of` and `$err`.
+
+Panic if:
+
+- A reserved compare mode is given
+- `$rA` is a [reserved register](./index.md#semantics)
+- `$rB + 32` overflows or `> VM_MAX_RAM`
+- `indirect == 1` and `$rC + 32` overflows or `> VM_MAX_RAM`
+
+### WDOP: Misc 128-bit integer operations
+
+|             |                                                                                       |
+|-------------|---------------------------------------------------------------------------------------|
+| Description | Perform an ALU operation on two 128-bit integers                                      |
+| Operation   | `b = mem[$rB,16];`<br>`c = indirect?mem[$rC,16]:$rC;`<br>`mem[$rA,16] = op(b,c);`     |
+| Syntax      | `wdop $rA, $rB, $rC, imm`                                                             |
+| Encoding    | `0x00 rA rB rC i`                                                                     |
+| Notes       |                                                                                       |
+
+The six-bit immediate value is used to select operating mode, as follows:
+
+Bits     | Short name | Description
+---------|------------|-------------
+`...XXX` | `op`       | Operation selection, see below
+`.XX...` | `reserved` | Reserved and must be zero
+`X.....` | `indirect` | Is rhs operand ($rC) indirect or not
+
+Then the actual operation that's performed:
+
+`op` | Name | Description
+-----|------|------------
+0    | add  | Add
+1    | sub  | Subtract
+2    | not  | Invert bits (discards rhs)
+3    | or   | Bitwise or
+4    | xor  | Bitwise exclusive or
+5    | and  | Bitwise and
+6    | shl  | Shift left (logical)
+7    | shr  | Shift right (logical)
+
+Operations behave `$of` and `$err` similarly to their 64-bit counterparts.
```

Comment:
Should we call out that `$of` in this case would simply be set to true, instead of the overflow result which would be the case for 64-bit addition?

Reply:
Well for addition the overflow is always either zero or one, since we can overflow one bit at most. However, bit shift right is affected by this. I'll update the comment.